pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -243,45 +207,19 @@ func (lb *ListenerBuilder) buildWaypointInboundTerminateConnect(svcs map[host.Na
 	return l
 }
 
-func (lb *ListenerBuilder) buildWaypointInboundOriginateConnect() *listener.Listener {
-	name := "inbound_CONNECT_originate"
-	l := &listener.Listener{
-		Name:              name,
-		UseOriginalDst:    wrappers.Bool(false),
-		ListenerSpecifier: &listener.Listener_InternalListener{InternalListener: &listener.Listener_InternalListenerConfig{}},
-		ListenerFilters:   []*listener.ListenerFilter{util.InternalListenerSetAddressFilter()},
-		FilterChains: []*listener.FilterChain{{
-			Filters: []*listener.Filter{{
-				Name: wellknown.TCPProxy,
-				ConfigType: &listener.Filter_TypedConfig{
-					TypedConfig: protoconv.MessageToAny(&tcp.TcpProxy{
-						StatPrefix:       name,
-						ClusterSpecifier: &tcp.TcpProxy_Cluster{Cluster: name},
-						TunnelingConfig: &tcp.TcpProxy_TunnelingConfig{
-							Hostname: "%DOWNSTREAM_LOCAL_ADDRESS%",
-							HeadersToAdd: []*core.HeaderValueOption{
-								{Header: &core.HeaderValue{Key: "x-envoy-original-dst-host", Value: "%DOWNSTREAM_LOCAL_ADDRESS%"}},
-							},
-						},
-					}),
-				},
-			}},
-		}},
-	}
-	return l
-}
-
-// VIP listeners, `inbound||hostname|port`. This will apply service policies. For typical case (not redirecting to external service),
-// this will end up forwarding to a cluster for the same VIP, which will have endpoints for each Pod internal listener
-func (lb *ListenerBuilder) buildWaypointInboundVIP(svcs map[host.Name]*model.Service) []*listener.Listener {
-	listeners := []*listener.Listener{}
+func (lb *ListenerBuilder) buildWaypointInternal(wls []WorkloadAndServices, svcs map[host.Name]*model.Service) *listener.Listener {
+	ipMatcher := &matcher.IPMatcher{}
```

Comment:
I am seeing 
```
2023-02-08T15:39:12.221608Z     warning envoy config external/envoy/source/common/config/grpc_subscription_impl.cc:128  gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) internal: Proto constraint validation failed (IPMatcherValidationError.RangeMatchers[0]: embedded message failed validation | caused by IPRangeMatcherValidationError.Ranges: value must contain at least 1 item(s)): range_matchers {
  on_match {
    action {
      name: "direct"
      typed_config {
        [type.googleapis.com/google.protobuf.StringValue] {
          value: "direct"
        }
      }
    }
  }
}
```

when there are zero pods

Reply:
Thanks, need to add a check for empty matcher in a follow-up.